### PR TITLE
Canada

### PR DESCRIPTION
**Pull request by mepa1363**
My apologies if this is a rather large PR. It enables the following:

- Adds locals for Canada.
- Adds gender ~“X”~ "Other" to represent LGBTQ2 community and to follow Canadian law.
- Changes the postal code pattern to be compatible with Canadian postal code format.
- Adds maximum zoom for the map as a config parameter. The reason is depending on the size of areas that represent postal codes you may want to limit the zoom so that the map does not mislead users with allowing to zoom in too much.
- ~Adds custom attribution as a config parameter.~ The base map uses data from OpenStreetMap so it has to be part of the attribution. ~Plus, you may get the postal code data from different sources so there should be a place to set the relevant info.~

### CODE REVIEW

File: app/config.ts [hunk 1]
```diff
@@ -13,6 +13,10 @@ const fallbackConfig: Config = {
   COUNTRY: 'Norway',
   MAP_CENTER: '10.7522, 63.9139',
   MAP_ZOOM: 4,
+  MAP_MAX_ZOOM: 13,
+  ZIP_DATA_ATTRIBUTION_TEXT: 'Postal code data',
+  ZIP_DATA_ATTRIBUTION_PROVIDER_NAME: 'GeoNames',
+  ZIP_DATA_ATTRIBUTION_PROVIDER_LINK: 'https://www.geonames.org/',
```
Comment (fossecode) [inline app/config.ts]: Do we need the attribution data in the config? Can't this just be placed in the text files instead?

File: app/domain/types.ts [hunk 2]
```diff
@@ -21,7 +21,8 @@ export type Symptoms = {
 
 export enum Sex {
   MALE = 'MALE',
-  FEMALE = 'FEMALE'
+  FEMALE = 'FEMALE',
+  X = 'X'
```
Comment (fossecode) [inline app/domain/types.ts]: Yeah it's probably time we add a third option here. Could it maybe be `OTHER` instead? We also need to add the text to all the existing locales. Not sure which text we should go for, "Other", "I prefer not to say", or something else

File: app/views/pages/report.ejs [hunk 3]
```diff
@@ -194,12 +194,15 @@
...
+      <label class="mr-4" for="gender-male"><%= __("Male") %></label>
+
+      <input required type="radio" id="gender-x" name="gender" value="x" <%= sex === 'X' ? 'checked' : '' %>>
+      <label for="gender-x"><%= __("X") %></label>
     </p>
     </fieldset>
 
     <label for="postal-code"><%= __('Zip code') %></label>
-    <input class="appearance-none border rounded py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" required type="text" pattern="[0-9]{<%= locals.zipLength %>}" maxlength="<%= locals.zipLength %>" name="postal-code" id="postal-code" placeholder="<%= locals.zipPlaceHolder %>" value="<%= postalCode %>">
+    <input class="appearance-none border rounded py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" required type="text" pattern="[A-Za-z][0-9][A-Za-z]" maxlength="<%= locals.zipLength %>" name="postal-code" id="postal-code" placeholder="<%= locals.zipPlaceHolder %>" value="<%= postalCode %>">
```
Comment (fossecode) [inline app/views/pages/report.ejs]: I suspect this change is going to break some existing languages. We have a couple of options:
1. Remove the pattern validation (don't know how much it helps)
2. Move the pattern to the config
3. Use a very wide pattern (`[A-Za-z0-9]{3,<%= locals.zipLength %>}`)

**Review by fossecode — CHANGES_REQUESTED** [review]
Look at comments above

File: app/config.ts (hunk 1 above)
Comment (fossecode) [reply app/config.ts]: (btw, the attribution that was there before is no longer relevant, so you can just place it directly there)

Comment (mepa1363) [reply app/config.ts]: I just removed the attribution as a config param and hardcoded it.

File: app/domain/types.ts (hunk 2 above)
Comment (mepa1363) [reply app/domain/types.ts]: I used "Other" instead of "X" to be more gender inclusive.

File: app/views/pages/report.ejs (hunk 3 above)
Comment (mepa1363) [reply app/views/pages/report.ejs]: I used option 3.

**Review by fossecode — APPROVED** [review]

**Comment by fossecode** [comment]
I added an extra comma when resolving the conflicts, I will fix it after merge

**Pull request merged**